app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -24,6 +24,10 @@ public function create(Request $request)
 
         $campaign = Campaign::with('characters', 'users')->where('code', $code)->first();
 
+        if (!$campaign) {
+            return redirect()->back()->withErrors(['error' => "The provided campaign code is incorrect."]);
+        }
```

Comment:
I don't see a test for this, I'm guessing since coverage isn't checked on this PR? Would probably only need 4-5 lines of code.

Reply:
Just added it :) good  catch!